crates/prune/prune/src/pruner.rs

```diff
@@ -50,44 +55,82 @@ pub struct Pruner<DB> {
     #[doc(hidden)]
     metrics: Metrics,
     event_sender: EventSender<PrunerEvent>,
+    state: S,
 }
 
-impl<DB: Database> Pruner<DB> {
-    /// Creates a new [Pruner].
+impl<DB> Pruner<WithoutProviderFactory, DB> {
+    /// Creates a new [Pruner] without a provider factory.
```

Comment:
`WithProviderFactory` is used when we initialize the pruner hook with the provider factory from the beginning https://github.com/paradigmxyz/reth/blob/232c6ae336decea09c5387caa736bd21a880d14f/crates/node/builder/src/launch/mod.rs#L242-L246 https://github.com/paradigmxyz/reth/blob/232c6ae336decea09c5387caa736bd21a880d14f/crates/consensus/beacon/src/engine/hooks/prune.rs#L101

`WithoutProviderFactory` is used in this stage, and the provider is passed directly to the `run` method, because we want to control the tx begin/commit outside the stage

Reply:
so this type is actually useless?

can flip the generics and use default = () type?